### PR TITLE
Add AppVeyor integration

### PR DESCRIPTION
**Pull request by davidebeatrici**
Build with #491: https://ci.appveyor.com/project/davidebeatrici/softethervpn/build/Build-1
Build without #491: https://ci.appveyor.com/project/davidebeatrici/softethervpn/build/Build-2

Commit ee1d84361efcccf480b3b4ab23a91a70429e2e3c adds support for [Visual Studio 2008 Express](https://stackoverflow.com/a/15319069/6658002), as the ATL/MFC resources are included in [Windows Driver Kit 7.1.0](https://www.microsoft.com/en-us/download/details.aspx?id=11800).

[Visual C++ 2008 Express](https://www.appveyor.com/docs/build-environment/#visual-studio-2008) and [Windows Driver Kit 7.1.0](https://www.appveyor.com/docs/build-environment/#windows-sdks) are pre-installed on both Visual Studio 2013 and Visual Studio 2015 images.

The LNK4099 warnings happen because the static libraries we have in the repository should have dedicated PDB files, but they are missing.

@dnobori Could you add these files to the repository, please?

---
SoftEther VPN Patch Acceptance Policy:
http://www.softether.org/5-download/src/9.patch

I choose option 1.

### CODE REVIEW

File: .appveyor.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+version: 'Build-{build}'
```
Comment (davidebeatrici) [inline .appveyor.yml]: This is the name of the build, where `{build}` is the incremental number.

I think that we should change it to something better, `Build-1` and `Build-2` don't seem to sound very well.

File: .appveyor.yml [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+pull_requests:
```
[the remark this answers is not in the document]
Comment (davidebeatrici) [reply .appveyor.yml]: Yes, Visual Studio 2015 is the default image.

I'm going to specify it explicitly.

File: .appveyor.yml (hunk 1 above)
Comment (davidebeatrici) [reply .appveyor.yml]: `{build}` is required.

What about `{build}-{commit-sha1}`?

**Comment by davidebeatrici** [comment]
Done.

**Comment by moatazelmasry2** [comment]
Cool. This time, I'd actually like to rebase these commits into the master branch instead of squash, since each commit is actually self contained and achieve one isolated task. Agree?

**Comment by davidebeatrici** [comment]
I strongly agree.

**Comment by moatazelmasry2** [comment]
Cool. Then please adapt the version in .appveyor.yml as agreed and we can merge :)

**Comment by davidebeatrici** [comment]
I replied to your comment, but the discussion is now marked as outdated as I pushed the changes.

>{build} is required.
>
>What about {build}-{commit-sha1}?

`4-1e94d9e`

https://ci.appveyor.com/project/davidebeatrici/softethervpn/build/4-1e94d9e

**Comment by davidebeatrici** [comment]
I see that you have created the project on AppVeyor, but the build failed.

That's because I decided to change the default configuration name `appveyor.yml` to `.appveyor.yml`.

You can set it in the project settings, under the `General` tab: "Custom configuration .yml file name".

**Comment by moatazelmasry2** [comment]
build-{build}-{commit} is fine

**Comment by moatazelmasry2** [comment]
Ah ok. then we can use my appveyor for the time being, until we agree on the others on how to actually manage this kind of stuff, in case me or Daiyuu died tomorrow :)

**Comment by davidebeatrici** [comment]
I changed `{build}-{commit}` to `build-{build}-{commit}`.
https://ci.appveyor.com/project/davidebeatrici/softethervpn/build/build-5-69347d3

When you login to AppVeyor via GitHub, you can choose between your personal profile and the organization one.

If you login with the organization one, you can create the project which will be accessible by everyone who is part of the team.

The URL will be https://ci.appveyor.com/project/softethervpn/softethervpn.

**Comment by moatazelmasry2** [comment]
It turned out that I should not login into appveyor using my github, instead create a new account using username/password then add authorize apveyor to use github

**Comment by moatazelmasry2** [comment]
I just rebased your changes and I'm really excited about them. Many thanks for your hard work

**Comment by davidebeatrici** [comment]
You're welcome!

**Comment by davidebeatrici** [comment]
The build failed, as expected: https://ci.appveyor.com/project/SoftEtherVPN/softethervpn/build/build-2-b29ca0b

Once #491 is merged, it should succeed.

**Pull request merged**